pg_search/src/postgres/customscan/pdbscan/mod.rs

```diff
@@ -1017,36 +1019,33 @@ fn choose_exec_method(privdata: &PrivateData) -> ExecMethodType {
             sort_direction,
             need_scores: privdata.need_scores(),
         }
-    } else {
-        if !fast_fields::fast_field_capable_prereqs(privdata) {
-            return ExecMethodType::Normal;
+    } else if fast_fields::is_numeric_fast_field_capable(privdata)
+        && gucs::is_fast_field_exec_enabled()
+    {
+        // Check for numeric-only fast fields first because they're more selective
+        ExecMethodType::FastFieldNumeric {
+            which_fast_fields: privdata.planned_which_fast_fields().clone().unwrap(),
         }
-        if fast_fields::is_numeric_fast_field_capable(privdata)
-            && gucs::is_fast_field_exec_enabled()
-        {
-            // Check for numeric-only fast fields first because they're more selective
-            ExecMethodType::FastFieldNumeric {
-                which_fast_fields: privdata.planned_which_fast_fields().clone().unwrap(),
-            }
-        } else if fast_fields::is_string_agg_capable(privdata).is_some()
-            && gucs::is_fast_field_exec_enabled()
-        {
-            let field = fast_fields::is_string_agg_capable(privdata).unwrap();
-            ExecMethodType::FastFieldString {
-                field,
-                which_fast_fields: privdata.planned_which_fast_fields().clone().unwrap(),
-            }
-        } else if fast_fields::is_mixed_fast_field_capable(privdata)
-            && gucs::is_mixed_fast_field_exec_enabled()
-        {
-            // Use MixedFastFieldExec if enabled
-            ExecMethodType::FastFieldMixed {
-                which_fast_fields: privdata.planned_which_fast_fields().clone().unwrap(),
-            }
-        } else {
-            // Fall back to normal execution
-            ExecMethodType::Normal
+    } else if fast_fields::is_string_agg_capable(privdata).is_some()
+        && gucs::is_fast_field_exec_enabled()
+    {
+        let field = fast_fields::is_string_agg_capable(privdata).unwrap();
+        ExecMethodType::FastFieldString {
+            field,
+            which_fast_fields: privdata.planned_which_fast_fields().clone().unwrap(),
         }
+    } else if gucs::is_mixed_fast_field_exec_enabled() {
+        // Use MixedFastFieldExec if enabled
+        //
+        // We'd suggest using MixedFastFieldExec as the last resort (default) at the planning
+        // stage, but we will fall back to NormalExecState (in assign_exec_method) if we can't
+        // execute using MixedFastFieldExec with the given fields and possibly expressions.
+        ExecMethodType::FastFieldMixed {
+            which_fast_fields: privdata.planned_which_fast_fields().clone().unwrap(),
+        }
```

Comment:
Nit: I still don't really love the "choose a method, but not really" approach, but this is well documented.

It would be great to continue to try to structure the code so that we don't fall back, as that is what is going to allow for giving useful estimates in the medium term.

Reply:
For example: we really ought to be claiming that the cost of a fast-field execution is `(num_fields * pg_sys::cpu_index_tuple_cost)` and that for Normal, it's `pg_sys::cpu_tuple_cost`.